src/core/Akka/Util/ByteString.cs

```diff
@@ -544,27 +450,17 @@ public int CopyTo(byte[] buffer, int index, int count)
             if (index < 0 || index >= buffer.Length) throw new ArgumentOutOfRangeException(nameof(index), "Provided index is outside the bounds of the buffer to copy to.");
             if (count > buffer.Length - index) throw new ArgumentException("Provided number of bytes to copy won't fit into provided buffer", nameof(count));
 
-            count = Math.Min(count, _count);
-            var remaining = count;
-            var position = index;
-            foreach (var b in _buffers)
-            {
-                var toCopy = Math.Min(b.Count, remaining);
-                Array.Copy(b.Array, b.Offset, buffer, position, toCopy);
-                position += toCopy;
-                remaining -= toCopy;
-
-                if (remaining == 0) return count;
-            }
-
-            return 0;
+            count = Math.Min(count, Count);
+            _memory.Span.CopyTo(buffer.AsSpan(index, count));
+            return count;
         }
 
         /// <summary>
         /// Copies content of a current <see cref="ByteString"/> into a provided <see cref="Memory{T}"/>
         /// <paramref name="buffer"/>
         /// </summary>
         /// <returns>The number of bytes copied</returns>
+        [Obsolete("This method will be removed in future versions of Akka.NET.")]
```

Comment:
butwhy.jpg?

(also same question for other more optimal overloads compared to the `byte[]` variant)

Reply:
It's duplicative - just use the methods on the `ReadOnlyMemory<byte>` property directly that already do this.

Reply:
Essentially, I'm trying to lower TCO here. .NET Standard 2.0 will be going bye-bye soon enough. We can get by just using the native System.Memory methods instead.